epidose/device/wps_scanner_d.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/sh
+#
+# Check for available networks and connect to the closest one using the WPS PBS
+#
+# Copyright 2020 Stefanos Georgiou
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -e
+
+export APP_NAME=wps_scanner_d
+
+# Location of suspend watchdod led
+SUSPEND_WATCHDOG_LED=/var/lock/epidose/suspend-watchdog-led
+
+# Pick up utility functions relative to the script's source code
+UTIL="$(dirname "$0")/util.sh"
+
+# Source common functionality (logging, WiFi)
+# shellcheck source=epidose/device/util.sh
+SERVER_URL="http://ha-server:5010" . "$UTIL"
+
+# Light up LED lights to indicate network status messages.
+# precondition: WiFi should be turned on.
+# Returns an exit code of 0 if a network connection is established,
+# while exit code 1 denotes a failed attempt to connect to a WPS network,
+# exit code 2 shows that there are no available WPS networks in the area,
+# and value 3 refers to an invalid LED color option.
+wps_led_blink()
+{
+  run_python device_io --"$1"-on
+  case "$1" in
+	"green") log "Obtained network connection"; return 0;;
+	"red") log "Failed to connect to a network"; return 1;;
+	"orange") log "No WPS network found in the area"; return 2;;
+	*) log "The given LED option is not available"return 3;;
+  esac
+  sleep 3
+  run_python device_io --"$1"-off
+}
+
+# Connect to a know network if available,
+# otherwise try to connect to the network with
+# the stongest signal by using WPS.
+# precondition: none
+# Returns exit codes based on the return codes
+# of the wps_led_blink function
+wps_connect()
+{
+  network_status=$(ifconfig wlan0 | grep "inet ")
+
+  # Check if device is connected to any network
+  if [ -z "$network_status" ] ; then
```

Comment:
I suggest to test for the opposite condition and return if the connection is there, to reduce the required nesting.

Reply:
Done.